extensions/ql-vscode/src/data-extensions-editor/bqrs.ts

```diff
@@ -37,9 +39,25 @@ export function decodeBqrsToExternalApiUsages(
       methodDeclaration.indexOf("("),
     );
 
+    // For Java, we'll always get back a .jar file, and the library version may be bad because not all library authors
+    // properly specify the version. Therefore, we'll always try to parse the name and version from the library filename
+    // for Java.
+    if (library.endsWith(".jar") || libraryVersion === "") {
```

Comment:
I'm not 100% on all the logic going on here, so just correct me if this is all fine and I just don't understand 🙇🏼 

I wonder if it's a good idea to be modifying `library` when `libraryVersion` is unknown. The `parseLibraryFilename` method always at least performs `basename` on the input, so if we couldn't determine the version we'll still perform some operation on the library name.

I wonder should we perhaps always be calling `parseLibraryFilename`? Or should we rename to `parseLibraryVersion` and change the logic so it only returns anything (and therefore we only modify `library` here) if it was able to split the library name into a name and version.

Reply:
The problem here is that we can get different information from the Java and C# queries. For Java, we get the JAR filename and a (potentially incorrect or missing) version. These are some examples of these tuples:

```
rt.jar, [empty string]
sql2o-1.6.0.jar, [empty string]
spring-boot-3.0.2.jar, [empty string]
mybatis-3.2.4.jar, 3.2.4
persistence-api-1.0.2.jar, 1.0
```

For a lot of libraries, the version is missing. In some libraries, the version is incorrect or missing some parts of the version. Therefore, we always need to parse the filename to check whether there is valid information there.

For C#, the tuples look like this:

```
System.Runtime, 4.2.2.0
Castle.Core, 5.0.0.0
Microsoft.Extensions.Logging.Abstractions, 3.1.9.0
```

That means that in C# we actually have the name of the assembly instead of the filename and have a correct version. Therefore, we shouldn't do any parsing in C#.

I think that means this logic makes sense:
- We shouldn't always parse the filename because that breaks things for C#.
- If the `parseLibraryFilename` function can't find a correct version in the filename, it will just return the filename without its extension. This would be the same as we would be doing if we only returned anything when we could parse the version, so we would be duplicating this logic.

Reply:
Thanks for explaining. I agree that considering the data we get from the queries it does make sense to do this processing in these cases.